kong/db/schema/others/declarative_config.lua

```diff
@@ -39,15 +39,33 @@ local foreign_references = {}
 local foreign_children = {}
 
 
-function DeclarativeConfig.pk_string(schema, object)
-  if #schema.primary_key == 1 then
-    return tostring(object[schema.primary_key[1]])
-  else
-    local out = {}
-    for _, k in ipairs(schema.primary_key) do
-      insert(out, tostring(object[k]))
+do
+  local tb_nkeys = require("table.nkeys")
+  local request_aware_table = require("kong.tools.request_aware_table")
```

Comment:
Is the usage of a `request_aware_table` necessary here? I feel like this operation is not bound to a request. Feel free to add some comments as to why this can't be a regular table and what exactly is CACHED_OUT

Reply:
@dndx , could you leave some comments?